reporter/internal/pdata/generate.go

```diff
@@ -195,8 +195,13 @@ func (p *Pdata) setProfile(
 	log.Debugf("Reporting OTLP profile with %d samples", profile.Sample().Len())
 
 	// Populate the deduplicated functions into profile.
-	for v := range funcMap {
-		f := profile.FunctionTable().AppendEmpty()
+	funcTable := profile.FunctionTable()
+	funcTable.EnsureCapacity(len(funcMap))
+	for range funcMap {
+		funcTable.AppendEmpty()
```

Comment:
It's unfortunate that the API of OTels `FunctionSlice` enforces these small memory/append operations and therefore increases the pressure on GC. As far as I can tell, this API needs to change, to get this fixed. I just noticed it here.

Reply:
I am not an expert in otel API, but  I was hoping `EnsureCapacity` would preallocate everything and then `AppendEmpty` just updates a slice bounds. Not 100% sure though.